src/vnsw/opencontrail-vrouter-netns/opencontrail_vrouter_netns/vrouter_hyperv.py

```diff
@@ -71,92 +155,72 @@ def __init__(self, vm_uuid, nic_left, nic_right, wingw_vm_name=None,
 
-        self.nic_left['name'] = (self.LEFT_DEV_PREFIX + "1")
-                                #self.nic_left['uuid'])[:self.NAME_LEN]
-        self.nic_right['name'] = (self.RIGHT_DEV_PREFIX + "2")
-                                #self.nic_right['uuid'])[:self.NAME_LEN]
-
-        self.nic_left['win_name'] = self.nic_left['win_name'][:self.NAME_LEN]
-        self.nic_right['win_name'] = self.nic_right['win_name'][:self.NAME_LEN]
-
-
-    def spawn_vm(self):
+    def spawn(self):
         """calls powershell to spawn vm """
-        if self._vm_exists():
+        if self._exists():
             raise ValueError("Specified Windows gateway VM already exists")
 
         self._configure_host_mgmt_ip()
 
-        powershell(["Copy-Item", self.vhd_path, self.cloned_disk_path])
+        call_powershell(["Copy-Item", self.vhd_path, self.cloned_disk_path])
 
-        powershell(["New-VM", "-Name", self.wingw_name, \
-                    "-Path", self.vm_location, \
-                    "-Generation", self.HYPERV_GENERATION, \
-                    "-MemoryStartupBytes", self.RAM_GB, \
-                    "-VHDPath", self.cloned_disk_path, \
-                    "-SwitchName", self.mgmt_vswitch_name])
+        call_powershell(["New-VM", "-Name", self.wingw_name, \
+                         "-Path", self.vm_location, \
+                         "-Generation", self.HYPERV_GENERATION, \
+                         "-MemoryStartupBytes", self.RAM_GB, \
+                         "-VHDPath", self.cloned_disk_path, \
+                         "-SwitchName", self.mgmt_vswitch_name])
 
-        powershell(["Set-VMFirmware", "-VMName", self.wingw_name, \
-                    "-EnableSecureBoot", "Off"])
+        call_powershell(["Set-VMFirmware", "-VMName", self.wingw_name, \
+                         "-EnableSecureBoot", "Off"])
 
-        powershell(["Start-VM", "-Name", self.wingw_name])
+        call_powershell(["Start-VM", "-Name", self.wingw_name])
 
         self.new_mgmt_ip = self._generate_new_mgmt_ip()
-        self._configure_vm_mgmt_ip()
-
-        powershell(["Add-VMNetworkAdapter", "-VMName", self.wingw_name,
-                    "-SwitchName", self.vrouter_vswitch_name,
-                    "-Name", self.nic_left['win_name']])
-        powershell(["Add-VMNetworkAdapter", "-VMName", self.wingw_name,
-                    "-SwitchName", self.vrouter_vswitch_name,
-                    "-Name", self.nic_right['win_name']])
+        self._configure_mgmt_ip()
 
+        call_powershell(["Add-VMNetworkAdapter", "-VMName", self.wingw_name,
+                         "-SwitchName", self.vrouter_vswitch_name,
+                         "-Name", self.nic_left.win_name])
+        call_powershell(["Add-VMNetworkAdapter", "-VMName", self.wingw_name,
+                         "-SwitchName", self.vrouter_vswitch_name,
+                         "-Name", self.nic_right.win_name])
 
-    def destroy_vm(self):
+    def destroy(self):
```

Comment:
I think that you should invoke this function when something went wrong also (because in other scenario we have to manually log into machine and delete/stop vm)